### PR TITLE
Add JsDoc 

### PR DESCRIPTION
**Pull request by IagoLast**
Add documentation for some expressions
---

There is a lot more to do, but as @rochoa said, always be closing, so I think we can merge this before adding more docs.

screen shot 2018-03-06 at 14 20 26

### CODE REVIEW

File: src/api/layer.js [hunk 1]
```diff
@@ -92,8 +87,8 @@ export default class Layer {
      * Set a new source for this layer.
      *
      * @param {carto.source.Base} source - New source
-     *
      * @memberof carto.Layer
+     * @instance
```
[the remark this answers is not in the document]
Comment (IagoLast) [reply src/api/layer.js]: otherwise is documented as `@static`

[the remark this answers is not in the document]
Comment (davidmanzanares) [reply src/api/layer.js]: :+1:

**Comment by IagoLast** [comment]
@davidmanzanares fixed, now it uses the defaut jsdoc theme and is ready to be integrated in the carto documentation center.

**Comment by IagoLast** [comment]
🚀

File: src/core/style/expressions/binary.js [hunk 2]
```diff
@@ -2,24 +2,328 @@ import { float } from '../functions';
...
+
+/**
+ *
+ * Compare if x is greater than or equal to y.
+ *
+ * This returns a numeric expression where 0 means `false` and 1 means `true`.
+ *
+ * @param {carto.style.expressions.Expression | number} x numeric expression
+ * @param {carto.style.expressions.Expression | number} y numeric expression
+ * @return {carto.style.expressions.Expression}
```
Comment (IagoLast) [inline src/core/style/expressions/binary.js]: I think we can assume that comparison operations will return a `float` expression

**Comment by IagoLast** [comment]
@davidmanzanares review of the recent changes.

- I agree with you about generalising the parameter types to `Expression` 
- We can asume the return type of some functions to be a `float` expression, I think expressions like eq, gt,  or floatSum will always return a `floatExpr`.

**Comment by davidmanzanares** [comment]
@IagoLast 

> We can assume the return type of some functions to be a float expression

We can't, they never are an instance of `Float` except in the `float()` case. The other functions, like `add()`, return a non-exported class that inherits from `Expression` and that is a numeric expression (`this.type='float'). However, that last part is an implementation detail. The user just needs to know that they are a numeric expression. I don't know if you were already referring to this.

**Pull request merged**